utilities/transactions/optimistic_transaction_test.cc

```diff
@@ -852,6 +852,37 @@ TEST_F(OptimisticTransactionTest, UntrackedWrites) {
   delete txn;
 }
 
+TEST_F(OptimisticTransactionTest, IteratorUpperBoundTest) {
+  WriteOptions write_options;
+  ReadOptions read_options;
+  auto txn = unique_ptr<Transaction>(txn_db->BeginTransaction(write_options));
+
+  string key1 = "a1";
+  string key2 = "a3";
+  string key3 = "b1";
+  string val = "123";
+  txn->Put(key1, val);
+  txn->Put(key2, val);
+
+  Status s = txn->Commit();
+  ASSERT_OK(s);
+  txn = unique_ptr<Transaction>(txn_db->BeginTransaction(write_options));
+  txn->Put(key3, val);
+
+  string ubKey("a2");
+  Slice upperbound(ubKey);
+  read_options.iterate_upper_bound = &upperbound;
+  auto it = unique_ptr<Iterator>(txn->GetIterator(read_options));
+  for (it->SeekToFirst(); it->Valid(); it->Next()) {
+    EXPECT_LT(it->key().ToString(), ubKey);
+  }
+  it.reset();
+
+  s = txn->Commit();
+  ASSERT_OK(s);
+  txn.reset();
+}
```

Comment:
Can you add a test case to cover that after Seek() it is already out of bound?

Reply:
Did you see this comment? We also should cover Seek() without out of bound for the first element.

Reply:
I guess I misunderstood your request, I added the second for loop to cover the case where a `SeekToFirst` call will make an invalid iterator valid again, I can add more test case to cover `Seek` with and without upper bound as well

Reply:
That is not what I mean. What I mean is that after Seek(), it is immediately Valid()=false because of out of bound.
The one you added is good one too.

Reply:
verification after Seek() call have been added, please take a look and let me know if it's sufficient.